### ISSUE TITLE
 Which #include <arm_math.h> for nano33BleSense CMSIS example

### ISSUE THREAD

**Issue by hpssjellis** [Fraunhofer-IMS/AIfES_for_Arduino#4]
On both the Portenta or the Nano33BleSense my Arduino IDE 1.18.15 compiler dies on the line 

#include <arm_math.h>

When I try to install it I only get the option to install the 

"Arduino_CMSIS-DSP"   which does not work

if I google arm_math I get several libraries such as:

https://github.com/ARM-software/CMSIS/blob/master/CMSIS/Include/arm_math.h

https://github.com/PaulStoffregen/arm_math

etc, etc. Any suggestions?

**Comment by AIfES-Pierre** [comment]
Hi Jeremy,

for the CMSIS example you definitely need the "Arduino_CMSIS-DSP" library. 

I just tested this successfully on the Portenta with the following setup:

- Arduino IDE version 1.8.13
- Board: Arduino Mbed Portenta Boards (Version 2.4.1)
- Arduino_CMSIS-DSP (Version 5.7.0)
- AIfES with the change in aifes.h as described in the manual.

Do you maybe still have the old Mbed core installed which is deprecated? You will only get the sketch compiled if the "Arduino_CMSIS-DSP" library is installed. This is still quite new and works only with the new Mbed cores. Seems to be a bug in the Arduino IDE. Maybe a reinstallation of the IDE will help. 

Many greetings
Pierre

**Comment by hpssjellis** [comment]
@AIfES-Pierre You are correct, this time installing the "Arduino_CMSIS-DSP" library and uncommenting ```#define AIFES_WITH_CMSIS``` from the AifES.h library source does seem to solve the issue I was having. Thank you. 

Surprised just adding the define into the sketch didn't solve the issue. I don't really want my students editing the library.


The following did not work, although I am not really sure why.
```
#define AIFES_WITH_CMSIS
#include <aifes.h>    

```

**Comment by AIfES-Pierre** [comment]
Hi @hpssjellis,

unfortunately this does not work. In the Arduino IDE the library is compiled first and then the sketch. So the define in the sketch has no effect at all. In addition, compiler/linker flags are not so easy to set in the Arduino IDE:
https://forum.arduino.cc/t/how-to-add-compile-options-to-arduino-ide-1-8-13/692640

That's why we solved it in such a complicated way. But we are working on a more elegant solution for the next update.

Many greetings
Pierre
